extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java

```diff
@@ -77,14 +90,76 @@ private void placeBreadcrumbFile( String folderPath, boolean success ) {
       logger.error( Messages.getInstance().getErrorString( "ActionInvokerSystemListener.ERROR_0004_COULD_NOT_WRITE_STATUS" ) );
     }
   }
+
+  private int pingIsAlive( ) {
+    try {
+      final HttpClient client = HttpClientManager.getInstance().createDefaultClient();
+      final URIBuilder uriBuilder = new URIBuilder()
+          .setScheme( HTTP_SCHEME )
+          .setHost( WN_HOST )
+          .setPort( WN_PORT )
+          .setPath( IS_ALIVE_ENDPOINT );
+      final URI pingIsAliveURI = uriBuilder.build();
+      final HttpGet pingIsAlive = new HttpGet( pingIsAliveURI.toString()  );
+      final HttpResponse response = client.execute( pingIsAlive );
+      return response.getStatusLine().getStatusCode();
+    } catch ( URISyntaxException | IOException e ) {
+      logger.error( e.getMessage() );
+    }
+    return HttpStatus.SC_INTERNAL_SERVER_ERROR;
+  }
+
+  private void waitUntilServerIsAlive( ) {
+    int timeout = 600000;
+    final int sleep_time = 5000;
+    boolean isAlive = false;
+    while ( !isAlive ) {
+      int statusCode = pingIsAlive();
+      if ( statusCode == HttpStatus.SC_OK ) {
+        isAlive = true;
+      } else {
+        if ( timeout == 0 ) {
+          placeBreadcrumbFile( getSolutionPath(), false );
```

Comment:
So we don't exit the while loop, but write the breadcrumb instead and that eventually leads to exiting the container, correct?